src/pages/QuestionPage.tsx

```diff
@@ -0,0 +1,235 @@
+import styled from '@emotion/styled';
+import { styleToken } from '@ui/styles';
+import { ChangeEvent, useEffect, useMemo, useState } from 'react';
+import { useNavigate } from 'react-router';
+import { Body } from '@ui/components/layout';
+import { BaseButton, NavigationHeader, Typography } from '@ui/components/common';
+import { useAxiosErrorAlert, useConfirm } from '@lib/hooks';
+import { ANSWER_TITLE } from '@lib/const/reportQnA';
+import { apiGetAnswers, apiGetQuestions, apiPostAnswer } from '../api/report';
+
+export function QuestionPage() {
+  const navigate = useNavigate();
+  const confirm = useConfirm();
+  const axiosErrorAlert = useAxiosErrorAlert();
+
+  const [questions, setQuestions] = useState<[]>([]);
+  const [answer, setAnswer] = useState<number[]>([]);
+
+  const isAvailable = useMemo(() => answer.length === 10, [answer]);
```

Comment:
P3:
```suggestion
  const isDisabledSubmit = useMemo(() => answer.length !== 10, [answer]);
```

Submit 버튼이 disabled되는 여부로 활용하려는 변수로 보여집니다.
isAvailable로 변수를 만들어주면 활용하는 곳에서는 한번더 반대로 값을 처리할 필요가 있을텐데요.
이럴때는 직관적이게 disabled 값으로 구성해주는것이 좋을것 같습니다.
그리고 변수명도 어느 부분에서 활용하는 것인지 명칭도 추가하여 직관적이게 개선해봤어요.

Reply:
해당 부분 개선하다가 마침 버그를 발견했어요. 10번 문항을 답변하면 [undefined * 9, 10번문항답변] 이런식으로 들어가서 무조건 버튼이 활성화 되더라구요. 

이 부분 filter를 이용해 유효한 값만 필터링하는 로직을 추가해 해결했습니다!

```
 const validAnswer = answer.filter((el) => el);
 const isDisabledSubmit = useMemo(() => validAnswer.length !== 10, [answer]);
```


<img width="368" alt="스크린샷 2023-10-05 오후 12 01 47" src="https://github.com/ShinjungOh/Haru-Hannyan/assets/97431021/18a1a9b9-12df-40c3-82bc-70e26b4266bc">
<img width="414" alt="스크린샷 2023-10-05 오후 12 03 25" src="https://github.com/ShinjungOh/Haru-Hannyan/assets/97431021/784276a1-b3dc-4bc3-9681-5c55b49f78f8">